### ISSUE TITLE
`CommandError`s and `AssertionErrors` when running test suite on `0.17.0`

### ISSUE THREAD

**Issue by TheChymera** [datalad/datalad#6832]
### What is the problem?

I tried running the new pytest suite and I get a lot of test failures:

```
========================================================= short test summary info ==========================================================
SKIPPED [1] datalad/metadata/extractors/tests/test_exif.py:24: unittest.case.SkipTest
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: Could not determine terminal size, skipping the rest of the test
SKIPPED [1] datalad/tests/utils_pytest.py:212: Module argcomplete fails to load
SKIPPED [5] datalad/tests/utils_pytest.py:336: Skip test known to fail
SKIPPED [49] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: Run this test by setting DATALAD_TESTS_SSH
SKIPPED [12] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: This test requires known credentials for datalad-test-s3
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: Core GitRepo class does not have format_commit() yet
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: No gin access token available
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: No gitea access token available
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: No github access token available
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: No gogs access token available
SKIPPED [7] datalad/tests/utils_pytest.py:846: No working HTTPS setup
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: Test requires `publish()` from datalad-deprecated
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: 'git annex init' with adjusted branch currently fails due to hard-coded 'origin'
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: provide tests for Keyring which interfaces keyring module
SKIPPED [1] datalad/tests/utils_pytest.py:336: run restricted cron due to rate limiting
SKIPPED [3] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: This test requires known credentials for crcns
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: need requests_ftp
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: Test can run only on yoh's setup
SKIPPED [3] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: This test requires known credentials for NDA
SKIPPED [1] datalad/tests/utils_pytest.py:336: git-annex version does not support `examinekey --migrate-to-backend`
SKIPPED [1] datalad/tests/utils_pytest.py:212: Module pyperclip fails to load
SKIPPED [2] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: Not fully tested/succeeded since some extractors failed to load:
No module named 'exifread'
No module named 'libxmp'
No module named 'exifread'
No module named 'libxmp'
SKIPPED [1] datalad/metadata/extractors/tests/test_xmp.py: Module 'libxmp' failed to load: No module named 'libxmp'
SKIPPED [2] datalad/tests/utils_pytest.py:212: Module libxmp fails to load
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: TODO: more, e.g. add with a custom backend
SKIPPED [2] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: TODO
SKIPPED [2] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: Cannot run sudo chown non-interactively: [Errno 2] No such file or directory: 'sudo'
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: External mvpa2 not present
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: TODO: file://::1/some does complain about parsed version dropping ::1
SKIPPED [1] ../../../../../../../usr/lib/python3.10/site-packages/_pytest/unittest.py:364: vcr is present, can't test behavior with vcr presence ATM
SKIPPED [1] datalad/tests/utils_pytest.py:336: condition was True
SKIPPED [1] datalad/tests/utils_pytest.py:212: Module line_profiler fails to load
XFAIL datalad/local/tests/test_rerun.py::test_run_inputs_outputs
  push fails due to IncompleteResultsError [remote rejected] (branch is currently checked out)
XFAIL datalad/resources/procedures/tests/test_noannex.py::test_noannex_fail_if_has_annexed
  Under pytest gets IncompleteResultsError with CommandError inside instead of actual CommandError
XPASS datalad/tests/test_utils.py::test_is_interactive TODO: for some reason fails on Travis
FAILED datalad/core/distributed/tests/test_clone.py::test_clone_crcns - AssertionError: Test 1/1: expected status ['ok'] not found in:
FAILED datalad/core/distributed/tests/test_clone.py::test_clone_datasets_root - datalad.support.exceptions.IncompleteResultsError: Comman...
FAILED datalad/core/distributed/tests/test_clone.py::test_autoenabled_remote_msg - datalad.support.exceptions.IncompleteResultsError: Com...
FAILED datalad/core/distributed/tests/test_clone.py::test_ria_http_storedataladorg - datalad.support.exceptions.IncompleteResultsError: C...
FAILED datalad/core/distributed/tests/test_clone.py::test_gin_cloning - datalad.support.exceptions.IncompleteResultsError: Command did no...
FAILED datalad/core/distributed/tests/test_clone.py::test_nonuniform_adjusted_subdataset - datalad.support.exceptions.IncompleteResultsEr...
FAILED datalad/distributed/tests/test_ria_basics.py::test_version_check - datalad.support.exceptions.IncompleteResultsError: Command did ...
FAILED datalad/distribution/tests/test_install.py::test_install_datasets_root - datalad.support.exceptions.IncompleteResultsError: Comman...
FAILED datalad/distribution/tests/test_install.py::test_install_simple_local[git] - datalad.support.exceptions.IncompleteResultsError: Co...
FAILED datalad/distribution/tests/test_install.py::test_install_simple_local[annex] - datalad.support.exceptions.IncompleteResultsError: ...
FAILED datalad/distribution/tests/test_install.py::test_install_dataset_from_just_source - datalad.support.exceptions.IncompleteResultsEr...
FAILED datalad/distribution/tests/test_install.py::test_install_dataset_from_just_source_via_path - datalad.support.exceptions.Incomplete...
FAILED datalad/distribution/tests/test_install.py::test_failed_install_multiple - AssertionError: assert ['ds1', 'ds3'] == ['crcns', 'ds1...
FAILED datalad/distribution/tests/test_install.py::test_datasets_datalad_org[] - datalad.support.exceptions.IncompleteResultsError: Comma...
FAILED datalad/distribution/tests/test_install.py::test_datasets_datalad_org[/.git] - datalad.support.exceptions.IncompleteResultsError: ...
FAILED datalad/distribution/tests/test_install.py::test_install_recursive_github - datalad.support.exceptions.NoDatasetFound: No installe...
FAILED datalad/downloaders/tests/test_s3.py::test_deny_access - socket.gaierror: [Errno -3] Temporary failure in name resolution
FAILED datalad/local/tests/test_download_url.py::test_download_url_need_datalad_remote - datalad.support.exceptions.IncompleteResultsErro...
FAILED datalad/support/tests/test_gitrepo.py::test_protocols[https] - datalad.runner.exception.CommandError: CommandError: 'git -c diff.i...
FAILED datalad/tests/test_installed.py::test_run_datalad_help - assert ['/usr/lib/python3.10/site-packages/requests/__init__.py:109: '\n ...
FAILED datalad/tests/test_misc.py::test_test - AttributeError: module 'datalad' has no attribute 'test'
FAILED datalad/tests/test_s3.py::test_get_versioned_url_anon - socket.gaierror: [Errno -3] Temporary failure in name resolution
FAILED datalad/tests/test_testrepos.py::test_clone - datalad.runner.exception.CommandError: CommandError: 'git clone https://github.com/d...
FAILED datalad/tests/test_utils_cached_dataset.py::test_get_cached_dataset - datalad.support.exceptions.IncompleteResultsError: Command d...
FAILED datalad/tests/test_utils_cached_dataset.py::test_cached_dataset - datalad.support.exceptions.IncompleteResultsError: Command did n...
FAILED datalad/tests/test_utils_cached_dataset.py::test_cached_url - datalad.support.exceptions.IncompleteResultsError: Command did not c...
====================== 26 failed, 1160 passed, 110 skipped, 2 xfailed, 1 xpassed, 5296 warnings in 6973.57s (1:56:13) ======================
```

### What steps will reproduce the problem?

Full build log: https://ppb.chymera.eu/9393aa.log

### Datalad information

datalad 0.17.0

```
# WTF
## configuration <SENSITIVE, report disabled by configuration>
## credentials
  - keyring:
    - active_backends:
      - EncryptedKeyring with [PBKDF2] AES256.CFB v.1.0 at /home/chymera/.local/share/python_keyring/crypted_pass.cfg
    - config_file: /home/chymera/.config/python_keyring/keyringrc.cfg
    - data_root: /home/chymera/.local/share/python_keyring
## datalad
  - version: 0.17.0
## dependencies
  - annexremote: 1.6.0
  - boto: 2.49.0
  - cmd:7z: 16.02
  - cmd:annex: 8.20200810
  - cmd:bundled-git: UNKNOWN
  - cmd:git: 2.35.1
  - cmd:ssh: 9.0p1
  - cmd:system-git: 2.35.1
  - cmd:system-ssh: 9.0p1
  - humanize: 4.2.3
  - iso8601: 1.0.2
  - keyring: 23.6.0
  - keyrings.alt: 4.1.0
  - msgpack: 1.0.4
  - mutagen: 1.45.1
  - platformdirs: 2.5.2
  - requests: 2.28.0
## environment
  - LANG: en_GB.utf8
  - PATH: /home/chymera/.local/bin:/home/chymera/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/bin:/usr/lib/llvm/14/bin
## extensions
## git-annex
  - build flags:
    - Assistant
    - Webapp
    - Pairing
    - S3
    - WebDAV
    - Inotify
    - DBus
    - DesktopNotify
    - TorrentParser
    - MagicMime
    - Feeds
    - Testsuite
  - dependency versions:
    - aws-0.22
    - bloomfilter-2.0.1.0
    - cryptonite-0.26
    - DAV-1.3.4
    - feed-1.3.0.1
    - ghc-8.10.4
    - http-client-0.6.4.1
    - persistent-sqlite-2.10.6.2
    - torrent-10000.1.1
    - uuid-1.3.13
    - yesod-1.6.1.0
  - key/value backends:
    - SHA256E
    - SHA256
    - SHA512E
    - SHA512
    - SHA224E
    - SHA224
    - SHA384E
    - SHA384
    - SHA3_256E
    - SHA3_256
    - SHA3_512E
    - SHA3_512
    - SHA3_224E
    - SHA3_224
    - SHA3_384E
    - SHA3_384
    - SKEIN256E
    - SKEIN256
    - SKEIN512E
    - SKEIN512
    - BLAKE2B256E
    - BLAKE2B256
    - BLAKE2B512E
    - BLAKE2B512
    - BLAKE2B160E
    - BLAKE2B160
    - BLAKE2B224E
    - BLAKE2B224
    - BLAKE2B384E
    - BLAKE2B384
    - BLAKE2BP512E
    - BLAKE2BP512
    - BLAKE2S256E
    - BLAKE2S256
    - BLAKE2S160E
    - BLAKE2S160
    - BLAKE2S224E
    - BLAKE2S224
    - BLAKE2SP256E
    - BLAKE2SP256
    - BLAKE2SP224E
    - BLAKE2SP224
    - SHA1E
    - SHA1
    - MD5E
    - MD5
    - WORM
    - URL
    - X*
  - operating system: linux x86_64
  - remote types:
    - git
    - gcrypt
    - p2p
    - S3
    - bup
    - directory
    - rsync
    - web
    - bittorrent
    - webdav
    - adb
    - tahoe
    - glacier
    - ddar
    - git-lfs
    - hook
    - external
  - supported repository versions:
    - 8
  - upgrade supported from repository versions:
    - 0
    - 1
    - 2
    - 3
    - 4
    - 5
    - 6
    - 7
  - version: 8.20200810
## location
  - path: /home/chymera
  - type: directory
## metadata_extractors
  - annex:
    - distribution: datalad 0.17.0
    - load_error: None
    - module: datalad.metadata.extractors.annex
  - audio:
    - distribution: datalad 0.17.0
    - load_error: None
    - module: datalad.metadata.extractors.audio
  - datacite:
    - distribution: datalad 0.17.0
    - load_error: None
    - module: datalad.metadata.extractors.datacite
  - datalad_core:
    - distribution: datalad 0.17.0
    - load_error: None
    - module: datalad.metadata.extractors.datalad_core
  - datalad_rfc822:
    - distribution: datalad 0.17.0
    - load_error: None
    - module: datalad.metadata.extractors.datalad_rfc822
  - exif:
    - distribution: datalad 0.17.0
    - load_error: ModuleNotFoundError(No module named 'exifread')
    - module: datalad.metadata.extractors.exif
  - frictionless_datapackage:
    - distribution: datalad 0.17.0
    - load_error: None
    - module: datalad.metadata.extractors.frictionless_datapackage
  - image:
    - distribution: datalad 0.17.0
    - load_error: None
    - module: datalad.metadata.extractors.image
  - xmp:
    - distribution: datalad 0.17.0
    - load_error: ModuleNotFoundError(No module named 'libxmp')
    - module: datalad.metadata.extractors.xmp
## metadata_indexers
## python
  - implementation: CPython
  - version: 3.10.5
## system
  - distribution: gentoo/2.8/n/a
  - encoding:
    - default: utf-8
    - filesystem: utf-8
    - locale.prefered: UTF-8
  - filesystem:
    - CWD:
      - max_pathlength: 4096
      - mount_opts: rw,noatime
      - path: /home/chymera
      - type: ext4
    - HOME:
      - max_pathlength: 4096
      - mount_opts: rw,noatime
      - path: /home/chymera
      - type: ext4
    - TMP:
      - max_pathlength: 4096
      - mount_opts: rw,noatime
      - path: /tmp
      - type: ext4
  - max_path_length: 269
  - name: Linux
  - release: 5.17.1-gentoo-r1
  - type: posix
  - version: #1 SMP PREEMPT Mon Apr 11 13:18:01 EDT 2022
  ```

### Have you had any success using DataLad before?

yes, I think the issue might be that some of these tests require dynamically downloading external data at build time. Ideally there would be a data package I can explicitly download, otherwise, perhaps there is something analogous to the `DATALAD_TESTS_NONETWORK=1` command previously used with nosetests?

**Comment by yarikoptic** [comment]
get 0.17.1 which addressed few of those, but you need to figure out at least one other in greater detail (not just summary that test failed due to some truncated message) since it is impossible for us to figure out what is specific to your setup.  FWIW also checkout https://github.com/datalad/datalad/blob/debian/debian/rules#L9 and recent changes there to match testing setup (switched to pytest from nose) which still has some rough edges

**Comment by yarikoptic** [comment]
> perhaps there is something analogous to the `DATALAD_TESTS_NONETWORK=1` command previously used with nosetests?

that variable still matters, use it!

**Comment by yarikoptic** [comment]
did you figure it all out @TheChymera so we could close this issue or there is something to address on datalad end?

**Comment by TheChymera** [comment]
@yarikoptic I think that solved it, additonally `0.17.0` had some issues with missing `allow_module_level=True` for pytest skipping, but I see that is fixed now.
